src/ghostfat.c

```diff
@@ -338,20 +396,27 @@ void uf2_read_block (uint32_t block_no, uint8_t *data)
       d->updateTime = COMPILE_DOS_TIME;
       d->updateDate = COMPILE_DOS_DATE;
       d->startCluster = startCluster & 0xFFFF;
-      d->size = (inf->content ? strlen(inf->content) : UF2_BYTE_COUNT);
+      d->size = (inf->content ? inf->size : UF2_BYTE_COUNT);
     }
   }
   else if ( block_no < BPB_TOTAL_SECTORS )
   {
     sectionRelativeSector -= FS_START_CLUSTERS_SECTOR;
-    uint32_t sectionRelativeClusterIndex = sectionRelativeSector / BPB_SECTORS_PER_CLUSTER;
 
-    if ( sectionRelativeClusterIndex < (NUM_FILES - 1) )
+    // plus 2 for first data cluster offset
+    uint32_t const cluster_num = 2 + sectionRelativeSector / BPB_SECTORS_PER_CLUSTER;
+
+    if ( cluster_num < UF2_FIRST_CLUSTER_NUMBER )
     {
-      // WARNING -- code presumes first data cluster == first file, second data cluster == second file, etc.
-      struct TextFile const * inf = &info[ sectionRelativeSector / BPB_SECTORS_PER_CLUSTER ];
-      size_t fileContentStartOffset = (sectionRelativeSector % BPB_SECTORS_PER_CLUSTER) * BPB_SECTOR_SIZE;
-      size_t fileContentLength = strlen(inf->content);
+      // Files that is not CURRENT.UF2
```

Comment:
grammar nit pick: choose one of:

`// Files that are not CURRENT.UF2"`

`// File that is not "CURRENT.UF2"`

I think the intent is:

`// Handle all files that are not "CURRENT.UF2"`

Reply:
fixed in my branch

Reply:
Right, I am terrible with grammar